### DISCUSSION TITLE
/x or /xq command line options not working when comparing folders?

### DISCUSSION THREAD

**Discussion by CarlColijn** [WinMerge/winmerge#640] [Ideas]
When I try to use the /x or /xq command line switch when comparing individual files this works wonderfully.  However, when I tell WinMerge to compare two identical folders these switches don't seem to work.  This is with and without a /r switch present.

Looking at the source I think I found the answer; the file `MergeDoc.cpp` has the following on line 2935:
    
    // Exit if files are identical should only work for the first
    // comparison and must be disabled afterward.
    theApp.m_bExitIfNoDiff = MergeCmdLineInfo::Disabled;

So I guess it's by design.  The bigger question however is: "Why", and my follow-up question would be: "Wouldn't it be nice if it worked though?"

**Comment by Cecil4th** [comment]
Hello. Feature request: Please have the command line options '/x' and '/xq' work the same for comparing folders as it does for comparing files. I am automating software testing with WinMerge, using exitcodes to determine pass/fail. The command line I use is: WinMergeU.exe /r /e /m Full /xq /u /enableexitcode "LEFT_ITEM" "RIGHT_ITEM". Currently WinMerge remains open awaiting user input when directory comparison is complete and closes with exitcode when file comparison is complete. WinMerge is excellent, I have been using it for many years, thank you all for maintaining and improving it. -Cecil

**Reply by sdottaka** [reply]
I copied this Discussion to an issue.
Although the behavior is slightly different from /xq, if you use the /noninteractive option, WinMerge will close immediately after completing a folder comparison. However, it will exit even if differences are found.

**Reply by Cecil4th** [reply]
Well thank you my new friend. I saw /noninteractive : TBD on the Command Line list (https://manual.winmerge.org/en/Command_line.html) and did not try it until just now. Works perfectly fine for my purposes, so this is no longer an issue.

All the best,
-Cecil
